src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/DependencyInjection/HostBuilderExtensions.cs

```diff
@@ -20,4 +24,18 @@ public static IFunctionsWorkerApplicationBuilder EnableMcpToolMetadata(this IFun
 
         return builder;
     }
-}
\ No newline at end of file
+
+    public static IFunctionsWorkerApplicationBuilder ConfigureMcpExtension(this IFunctionsWorkerApplicationBuilder builder)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        builder.UseMiddleware<FunctionsMcpContextMiddleware>();
```

Comment:
Does this need a condition? It looks like this middleware will be run against any invocation. I see we effectively guard against that in `TryGetMcpToolTriggerName`, but I wonder if a `UseWhen` going through the `FunctionDefinition` would be a more efficient circuit break.